src/Bitcoin/TransactionSigner.cpp

```diff
@@ -273,9 +273,11 @@ Data TransactionSigner<Transaction, TransactionBuilder>::pushAll(const std::vect
 template <typename Transaction, typename TransactionBuilder>
 Data TransactionSigner<Transaction, TransactionBuilder>::keyForPublicKeyHash(const Data& hash) const {
     for (auto& key : input.private_key()) {
-        auto publicKey = PrivateKey(key).getPublicKey(TWPublicKeyTypeSECP256k1);
-        auto keyHash = TW::Hash::ripemd(TW::Hash::sha256(publicKey.bytes));
-        if (keyHash == hash) {
+        auto pubKeyExtended = PrivateKey(key).getPublicKey(TWPublicKeyTypeSECP256k1Extended);
+        auto pubKey = pubKeyExtended.compressed();
+        auto keyHash = Hash::sha256ripemd(pubKey.bytes.data(), pubKey.bytes.size());
+        auto keyHash2 = Hash::sha256ripemd(pubKeyExtended.bytes.data(), pubKeyExtended.bytes.size());
```

Comment:
Extended keyhash computation can be skipped if non-extended keyHash matches (minor optimization)

Reply:
should be fixed